pkg/build/util/generate.go

```diff
@@ -40,28 +45,36 @@ func GenerateBuildFromBuild(build *api.Build) *api.Build {
 	}
 }
 
+// TODO will implement this in a future pull, want to get the refactor delivered before
+// too much rebasing happens.
+// CreateBuildWithImmutableID creates a build using the OpenShift client and the current immutable imageid
+// from any ImageRepository that is associated to the BuildConfig by an ImageChangeTrigger
+func GenerateBuildWithImmutableID(config *buildapi.BuildConfig) *buildapi.Build {
```

Comment:
at the time it was easier to leave it in w/ the refactoring i'd done, now i'll pull it out.